common/flatpak-utils-http.c

```
@@ -231,6 +243,159 @@ check_http_status (guint status_code,
+
+      enumerator = g_file_enumerate_children (host_dir, G_FILE_ATTRIBUTE_STANDARD_NAME,
+                                              G_FILE_QUERY_INFO_NONE,
+                                              NULL, &local_error);
+      if (enumerator == NULL)
+        {
+          /* This matches libpod - missing certificate directory or a permission
+           * error causes the directory to be skipped; any other error is fatal
+           */
+          if (g_error_matches(local_error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND) ||
+              g_error_matches(local_error, G_IO_ERROR, G_IO_ERROR_PERMISSION_DENIED))
+            {
+              g_clear_error (&local_error);
+              continue;
+            }
+          else
+            {
+              g_propagate_error (error, g_steal_pointer (&local_error));
+              return NULL;
+            }
+        }
+
+      certificates = g_new0 (FlatpakCertificates, 1);
+
+      while (TRUE)
+        {
+          GFile *child;
+          g_autofree char *basename = NULL;
+
+          if (!g_file_enumerator_iterate (enumerator, NULL, &child, NULL, error))
+            return NULL;
+
+          if (child == NULL)
+            break;
+
+          basename = g_file_get_basename (child);
+
+          /* In libpod, all CA certificates are added to the CA certificate
+           * database. We just use the first in readdir order.
+           */
+          if (g_str_has_suffix (basename, ".crt") && certificates->ca_cert_file == NULL)
+            certificates->ca_cert_file = g_file_get_path (child);
+
+          if (g_str_has_suffix (basename, ".cert"))
+            {
+              g_autofree char *nosuffix = g_strndup (basename, strlen (basename) - 5);
+              g_autofree char *key_basename = g_strconcat (nosuffix, ".key", NULL);
+              g_autoptr(GFile) key_file = g_file_get_child (host_dir, key_basename);
+
+              if (!g_file_query_exists (key_file, NULL))
```

Do we really need to report an error if we already chose the cert to use tho? (i.e. if certificates->client_cert_file != NULL already).

---

Docker does report those errors for all files. Yes, we don't have to, but to catch miss-configurations and be more compatible with docker, I think this is a good idea.